src/panels/developer-tools/template/developer-tools-template.ts

```diff
@@ -316,15 +328,33 @@ class HaPanelDevTemplate extends LitElement {
           white-space: pre-wrap;
           background-color: var(--secondary-background-color);
           padding: 8px;
+          margin-top: 0;
+          margin-bottom: 0;
           direction: ltr;
+          overflow: auto;
+        }
+        
+        p,
+        ul {
+          margin-block-end: 0;
+        }
+        
+        .content.horizontal .render-pane .card-content {
+          display: flex;
+          flex-direction: column;
+          max-height: calc(
+            100vh - var(--header-height) - var(--paper-font-body1_-_line-height) * 3 - 1em * 2
+            - max(16px,env(safe-area-inset-top)) * 2 - max(16px,env(safe-area-inset-bottom)) * 2
+            - var(--ha-card-border-width,1px) * 2 - 132px
+          );
```

Comment:
```suggestion
          overflow: auto;
           max-height: var(--code-mirror-max-height);
        }
        
        p,
        ul {
          margin-block-end: 0;
        }
        
        .content.horizontal .render-pane .card-content {
          display: flex;
          flex-direction: column;
```

Reply:
If I remove `max-height` from `.content.horizontal .render-pane .card-content` and set `max-height: var(--code-mirror-max-height);` for `.rendered` - then both panes will go down beyond a viewport.
Currently both panes fit in a viewport (with a small padding 16px below).

Reply:
> Please add height: 100% to .edit-pane -> panes are not same height and additional text can be out of view:

This is very strange, I do not see it...
![изображение](https://github.com/user-attachments/assets/fe5d4c5b-1094-4008-87c7-2d761bd039c1)
Will keep testing.

Reply:
> You need to rebase your branch to be able to fix prettier.

Rebasing did not help(

Reply:
> If I remove `max-height` from `.content.horizontal .render-pane .card-content` and set `max-height: var(--code-mirror-max-height);` for `.rendered` - then both panes will go down beyond a viewport. Currently both panes fit in a viewport (with a small padding 16px below).

Yeah but in my oppinion this is fine, because we want to have a full overview of our template and preview, we don't need to see on the first sight to which events the tempate is listening.

Reply:
Buttons will be shown, just the additional `this template listens for....` will need be be scrolled:

[Screencast from 2024-11-20 09-09-01.webm](https://github.com/user-attachments/assets/85fe2c87-b793-42ef-accf-eea2fe1408cb)